src/passes/Print.cpp

```diff
@@ -2346,6 +2346,33 @@ struct PrintSExpression : public UnifiedExpressionVisitor<PrintSExpression> {
       o << " ;; end try";
     }
   }
+  void printUnreachableReplacement(Expression* curr) {
+    // See the parallel function in PrintExpressionContents for background.
+    //
+    // Emit a block with drops of the children.
+    o << "(block";
+    incIndent();
+    for (auto* child : ChildIterator(curr)) {
+      Drop drop;
+      drop.value = child;
+      printFullLine(&drop);
+    }
+    decIndent();
+  }
+  void visitStructSet(StructSet* curr) {
+    if (curr->ref->type == Type::unreachable) {
+      printUnreachableReplacement(curr);
+      return;
+    }
+    visitExpression(curr);
+  }
+  void visitStructGet(StructGet* curr) {
+    if (curr->ref->type == Type::unreachable) {
+      printUnreachableReplacement(curr);
+      return;
+    }
+    visitExpression(curr);
+  }
```

Comment:
Aren't these handled by `PrintExpressionContents::visitStructGet` and `PrintExpressionContents::visitStructSet`? The call to `visitExpression` will call them I think..

Reply:
We have to override the default `visitExpression` here, in the case of unreachable children, so that we can fully handle that case, using `printUnreachableReplacement` to print the children with drops, and enclose them with a block. Otherwise we don't know how to print the Struct operation without an RTT.

Print `PrintExpressionContents` does something similar, but it doesn't print children, it just just prints the contents of the current node. So the code there is not enough, without the new code here we can't print the children properly.